src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,197 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> tail;
+    private Node<T> head;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (head == null) {
+            head = tail = new Node<>(null, value, null);
+        } else {
+            tail = tail.next = new Node<>(tail, value, null);
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        addByIndexRoutine(value, index);
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T t : list) {
+            add(t);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkSetIndexInBound(index);
+        return getNode(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        return setValueRoutine(value, index);
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        return unlinkNodeByIndex(index);
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        return unlinkNodeByObject(object);
```

Comment:
same here, there is no need to encapsulate each method's logic into a private methods, if you are thinking about reusage - we can reuse them by calling public methods as well